### ISSUE TITLE
Suggestion: Automod, prevent user from reacting too many times for set of period

### ISSUE THREAD

**Issue by maidos** [jonas747/yagpdb#485]
So there are new ways to spam our server and currently yagbot doesnt have any way to prevent this type of spam. 

Theres selfbot that react with specific word, usually racial and hateful letters and spam it accross multiple channels on several messages. 

I would suggest adding a feature to monitor the amount of reaction emotes a user can add in a set of period

like user can react emote x many times in y seconds.

**Comment by noahggarrett** [comment]
I agree I think this would be useful to prevent users from reacting to much and spamming the reaction logs channel I setup

**Comment by noahggarrett** [comment]
I think that sound like a great idea, but I don;t see many servers where that would be usefull

**Comment by Aldiwildan77** [comment]
I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.

**Comment by TheHDCrafter** [comment]
> I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.

Use custom commands like i do and some kind of a counting / violation system that makes it so reactionens are removed again .-.

As example i am pausing my custom reaction commands (giving a role and setting it to be ignored) so the custom reaction commands don't get flodded if people spam react ![Screenshot_20201111-185701.jpg](https://user-images.githubusercontent.com/29355595/98846820-c91bf100-244f-11eb-9c8a-48d6c5ee5256.jpg)

**Comment by Aldiwildan77** [comment]
> > I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.
> 
> Use custom commands like i do and some kind of a counting / violation system that makes it so reactionens are removed again .-.
> 
> As example i am pausing my custom reaction commands (giving a role and setting it to be ignored) so the custom reaction commands don't get flodded if people spam react ![Screenshot_20201111-185701.jpg](https://user-images.githubusercontent.com/29355595/98846820-c91bf100-244f-11eb-9c8a-48d6c5ee5256.jpg)

I've set the role and setting it to be ignored from creating a reaction. but if I do a reaction as an admin some peoples still have access to react my reaction. I want it still only me to give a reaction. have you think about that, I think it's weird for you.

**Comment by TheHDCrafter** [comment]
> > > I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.
> > 
> > Use custom commands like i do and some kind of a counting / violation system that makes it so reactionens are removed again .-.
> > 
> > As example i am pausing my custom reaction commands (giving a role and setting it to be ignored) so the custom reaction commands don't get flodded if people spam react ![Screenshot_20201111-185701.jpg](https://user-images.githubusercontent.com/29355595/98846820-c91bf100-244f-11eb-9c8a-48d6c5ee5256.jpg)
> 
> I've set the role and setting it to be ignored from creating a reaction. but if I do a reaction as an admin some peoples still have access to react my reaction. I want it still only me to give a reaction. have you think about that, I think it's weird for you.

Why not just create a custom command that auto unreacts again for the user?

**Comment by Aldiwildan77** [comment]
> > > > I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.
> > > 
> > > 
> > > Use custom commands like i do and some kind of a counting / violation system that makes it so reactionens are removed again .-.
> > > As example i am pausing my custom reaction commands (giving a role and setting it to be ignored) so the custom reaction commands don't get flodded if people spam react
> > 
> > 
> > I've set the role and setting it to be ignored from creating a reaction. but if I do a reaction as an admin some peoples still have access to react my reaction. I want it still only me to give a reaction. have you think about that, I think it's weird for you.
> 
> Why not just create a custom command that auto unreacts again for the user?

I think it needs an automoderator v2 to handle events for who's allowed or not allowed to give a reaction.

**Comment by TheHDCrafter** [comment]
> > > > > I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.
> > > > 
> > > > 
> > > > Use custom commands like i do and some kind of a counting / violation system that makes it so reactionens are removed again .-.
> > > > As example i am pausing my custom reaction commands (giving a role and setting it to be ignored) so the custom reaction commands don't get flodded if people spam react
> > > 
> > > 
> > > I've set the role and setting it to be ignored from creating a reaction. but if I do a reaction as an admin some peoples still have access to react my reaction. I want it still only me to give a reaction. have you think about that, I think it's weird for you.
> > 
> > Why not just create a custom command that auto unreacts again for the user?
> 
> I think it needs an automoderator v2 to handle events for who's allowed or not allowed to give a reaction. 

By the way for suggestions you can always join the [yagpdb support server](https://discord.gg/4uY54rw) and use -suggest there to request this feature

**Comment by Aldiwildan77** [comment]
> > > > > > I agree with this, such a good issue. some cases will be good if I want to validate a message with a reaction and so other people can't give a reaction because it should be only 1 reaction.
> > > > > 
> > > > > 
> > > > > Use custom commands like i do and some kind of a counting / violation system that makes it so reactionens are removed again .-.
> > > > > As example i am pausing my custom reaction commands (giving a role and setting it to be ignored) so the custom reaction commands don't get flodded if people spam react
> > > > 
> > > > 
> > > > I've set the role and setting it to be ignored from creating a reaction. but if I do a reaction as an admin some peoples still have access to react my reaction. I want it still only me to give a reaction. have you think about that, I think it's weird for you.
> > > 
> > > 
> > > Why not just create a custom command that auto unreacts again for the user?
> > 
> > 
> > I think it needs an automoderator v2 to handle events for who's allowed or not allowed to give a reaction.
> 
> By the way for suggestions you can always join the [yagpdb support server](https://discord.gg/4uY54rw) and use -suggest there to request this feature

okay, I'll make it to suggestion. thank you.

**Comment by NaruDevnote** [comment]
The size of these comments are astronomical.
